packages/firestore/test/unit/specs/listen_spec.test.ts

```diff
@@ -78,6 +78,19 @@ describeSpec('Listens:', [], () => {
     }
   );
 
+  specTest('Does not raise event for initial document delete', [], () => {
+    const query = Query.atPath(path('collection'));
+    const missingDoc = deletedDoc('collection/a', 1000);
+    return spec()
+      .userListens(query)
+      .watchAcks(query)
+      .watchSends({ removed: [query] }, missingDoc)
```

Comment:
Hrm...  why is watch removing a document from the target that was never included?

Reply:
This test basically follows the request trace I got from the validation test:

```
addTarget: 
   { documents: 
      { documents: 
         [ 'projects/node-review-with-fix/databases/(default)/documents/test/V9w4ini0kixUMxshKuhG' ] },
     targetId: 4 } }
Firestore (5.0.2) [Connection]: GRPC stream received: { response_type: 'targetChange',
  targetChange: 
   { targetChangeType: 'ADD',
     targetIds: [ 4 ],
     cause: null,
     resumeToken: <Buffer >,
     readTime: null },
  documentChange: null,
  documentDelete: null,
  documentRemove: null,
  filter: null }
Firestore (5.0.2) [Connection]: GRPC stream received: { response_type: 'documentDelete',
  targetChange: null,
  documentChange: null,
  documentDelete: 
   { document: 'projects/node-review-with-fix/databases/(default)/documents/test/V9w4ini0kixUMxshKuhG',
     removedTargetIds: [ 4 ],
     readTime: { seconds: '1526067903', nanos: 787493000 } },
  documentRemove: null,
  filter: null }
```

My resident Watch expert says "there is a good reason for this, but I don't remember".

Reply:
So I didn't realize Watch does this...  Maybe add a comment to make it clear that watch really does this and so you're testing this intentionally.

// To indicate the document doesn't exist, watch sends a DocumentDelete message as if the document previously existed and now is being deleted/removed from the target.

Reply:
Added.